pkg/sdkclient/sourcetransformer/client.go

```diff
@@ -53,18 +57,76 @@ func New(serverInfo *serverinfo.ServerInfo, inputOptions ...sdkclient.Option) (C
 	c := new(client)
 	c.conn = conn
 	c.grpcClt = transformpb.NewSourceTransformClient(conn)
+
+	var logger = logging.FromContext(ctx)
+
+waitUntilReady:
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("waiting for transformer gRPC server to be ready: %w", context.Cause(ctx))
+		default:
+			_, err := c.IsReady(ctx, &emptypb.Empty{})
+			if err != nil {
+				logger.Warnf("Transformer server is not ready: %v", err)
+				time.Sleep(100 * time.Millisecond)
+				continue waitUntilReady
+			}
+			break waitUntilReady
+		}
+	}
+
+	c.stream, err = c.grpcClt.SourceTransformFn(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("creating a gRPC stream for source transform: %w", err)
```

Comment:
I've seen somewhere (Go blog / talks), to not include "failed" in an error message, since `error` type itself indicates that it failed. If the error gets wrapped again on the caller's side, it could result in a final message of below pattern:
```
Failed to create user: Failed to create new account: Failed to INSERT new account record: database not connected
```

Compared to:
```
Failed to create user: creating new account:  INSERT new account record: database not connected
```